nodes/move-object/move-object.js

```diff
@@ -0,0 +1,153 @@
+      return;
+    }
+
+    this.on("input", async function (msg, send, done) {
+      let bucket = n.bucket !== "" ? n.bucket : null; // Bucket info
+      let key = n.key !== "" ? n.key : null; // Object key
+
+      let sourcekey = n.sourcekey !== "" ? n.sourcekey : null;
+      let sourcebucket = n.sourcebucket !== "" ? n.sourcebucket : null;
+
+      // Checking for correct properties input
+      if (!bucket) {
+        bucket = msg.bucket ? msg.bucket : null;
+        if (!bucket) {
+          node.error("No bucket provided!");
+          return;
+        }
+      }
+
+      if (!key) {
+        key = msg.key ? msg.key : null;
+        if (!key) {
+          node.error("No object key provided!");
+          return;
+        }
+      }
+
+      if (!sourcebucket) {
+        sourcebucket = msg.sourcebucket ? msg.sourcebucket : null;
+        if (!sourcebucket) {
+          node.error("No sourcebucket provided!");
+          return;
+        }
+      }
+
+      if (!sourcekey) {
+        sourcekey = msg.sourcekey ? msg.sourcekey : null;
+        if (!sourcekey) {
+          node.error("No sourcekey provided!");
+          return;
+        }
+      }
+
+      // S3 client init
+      let s3Client = null;
+
+      try {
+        // Creating S3 client
+        s3Client = new S3({
+          endpoint: config.endpoint,
+          forcePathStyle: config.forcepathstyle,
+          region: config.region,
+          credentials: {
+            accessKeyId: config.credentials.accesskeyid,
+            secretAccessKey: config.credentials.secretaccesskey,
+          },
+        });
+
+        // Uploading
+        node.status({
+          fill: "blue",
+          shape: "dot",
+          text: "Uploading",
+        });
+
+        s3Client.copyObject(
+          {
+            CopySource: sourcebucket + "/" + sourcekey,
+            Bucket: bucket,
+            Key: key,
+          },
+          function (err, data) {
+            if (err) {
+              // Uploading
+              node.status({
+                fill: "red",
+                shape: "dot",
+                text: "Failure",
+              });
+
+              node.error(err);
+            } else {
+              s3Client.deleteObject(
+                { Bucket: sourcebucket, Key: sourcekey },
+                function (err, data) {
+                  if (err) {
+                    node.status({ fill: "red", shape: "dot", text: `Failure` });
+                    node.error(err);
+                    // Replace the payload with null
+                    msg.payload = null;
+                    // Append the delete object
+                    // key to the message object
+                    msg.key = key;
+
+                    // Return the complete message object
+                    send(msg);
+                  } else {
+                    // Replace the payload with
+                    // the returned data
+                    msg.payload = data;
```

Comment:
I believe that the payload should be something else than the deletion data. I would say it's better to have the data from the `copyObject` function would be better since that response contains the newly created object's `ETag` and `VersionId`